src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -157,18 +160,53 @@ public void arcadeDrive(double throttle, double turnModifier){
     differentialDrive.arcadeDrive(throttle, turnModifier);
   }
 
+  /**
+   * Executes curvatureDrive with closed loop control. Needs to be called
+   * periodically by the command to update the closed loop. 
+   * 
+   * The closed loop works on the difference between the desired curvature
+   * and the measured curvature, using proportional gain only.
+   * 
+   * @param throttle - [-1.0, 1.0]. Positive is forward
+   * @param curvature - [-1.0, 1.0]. Positive is turning right
+   */
   public void curvatureDrive(double throttle, double curvature) {
-    if(throttle != 0.0 && curvature == 0.0 && !gyroDisabled) { //driving straight and no turn
-      if(this.desiredAngle == Integer.MAX_VALUE) { //means robot just started driving straight
-        this.desiredAngle = gyroscope.getGyroAngle(); 
-      }
-      curvature = this.getAngularError(desiredAngle) * P_VALUE; 
-      this.setLeftPower(throttle - curvature);
-      this.setRightPower(throttle + curvature);
-    }
-    else { // when robot isn't driving straight
-      this.desiredAngle = Integer.MAX_VALUE; //if turn is greater than 0 or if robot is still
-      differentialDrive.curvatureDrive(-throttle, curvature, (Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE));
+
+    if(Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE){
+      // If throttle is less than the deadzone, treat inputs as arcade
+      this.arcadeDrive(throttle, curvature);
+    }else{
+      // Otherwise, treat inputs as curvature
+
+      // Calculate ideal left and right speeds based on curvature
+      WheelSpeeds desiredSpeeds = DifferentialDrive.curvatureDriveIK(throttle, curvature, false);
```

Comment:
```suggestion
      WheelSpeeds desiredSpeeds = DifferentialDrive.curvatureDriveIK(throttle, curvature, false);
      desiredSpeeds.left = desiredSpeeds.left * DriveConstants.MAX_SPEED;
      desiredSpeeds.right =  desiredSpeeds.right * DriveConstants.MAX_SPEED;
      
      ChassisSpeeds chassisSpeeds = DriveConstants.kDriveKinematics.toWheelSpeeds(new DifferentialDriveWheelSpeeds(desiredSpeeds.left, desiredSpeeds.right);
```

Now we can do things like
`chassisSpeeds.vxMetersPerSecond` for desired linear velocity
or `chassisSpeeds.omegaRadiansPerSecond` for desired angular velocity which should simplify the math below. We will also be able to set a max speed on our robot.